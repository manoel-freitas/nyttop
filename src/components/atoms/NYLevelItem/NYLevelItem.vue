<script>
export default {
  props: {
    side: {
      type: String,
      requried: false,
      default: ''
    }
  },
  computed: {
    levelSide() {
      switch (this.side) {
        case 'right':
          return 1
        case 'left':
          return -1

        default:
          return 0
      }
    }
  },
  render() {
    const levelItem = (
      <div class={{ 'level-item': true }}>{this.$slots.default}</div>
    )
    const levelSideLeft = <div class={{ 'level-left': true }}>{levelItem}</div>
    const levelSideRight = (
      <div class={{ 'level-right': true }}>{levelItem}</div>
    )
    switch (this.side) {
      case 'right':
        return levelSideRight
      case 'left':
        return levelSideLeft

      default:
        return levelItem
    }
  }
}
</script>

<style></style>
