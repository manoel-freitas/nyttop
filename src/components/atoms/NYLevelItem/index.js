import NYLevelItem from './NYLevelItem'

export default NYLevelItem
