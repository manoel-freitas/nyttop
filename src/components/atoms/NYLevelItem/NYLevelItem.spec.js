/// <reference types="jest" />

import { mount, shallowMount } from '@vue/test-utils'
import NYLevelItem from './NYLevelItem'

test('NYLevelItem has to be mounted', () => {
  const nyLevelItem = mount(NYLevelItem)
  expect(nyLevelItem.isVueInstance()).toBeTruthy()
})

test('snapshot test default comp', () => {
  const nyLevelItem = mount(NYLevelItem)
  expect(nyLevelItem.element).toMatchSnapshot()
})
test('snapshot test right side comp', () => {
  const nyLevelItem = shallowMount(NYLevelItem)
  nyLevelItem.setProps({ side: 'right' })
  expect(nyLevelItem.vm.levelSide).toBe(1)
})
test('snapshot test default side comp', () => {
  const nyLevelItem = shallowMount(NYLevelItem)
  expect(nyLevelItem.vm.levelSide).toBe(0)
})

test('snapshot test left side comp', () => {
  const nyLevelItem = shallowMount(NYLevelItem)
  nyLevelItem.setProps({ side: 'left' })
  expect(nyLevelItem.vm.levelSide).toBe(-1)
})
