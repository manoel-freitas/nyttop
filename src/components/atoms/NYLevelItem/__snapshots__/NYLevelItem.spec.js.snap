// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test default comp 1`] = `
<div
  class="level-item"
/>
`;
